src/System.Net.Http/src/System/Net/Http/HttpMethod.cs

```diff
@@ -1,27 +1,24 @@
 // Copyright (c) Microsoft. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
-using System;
-using System.Collections.Generic;
-using System.Text;
-using System.Threading.Tasks;
-using System.Diagnostics.Contracts;
+using System.Diagnostics;
 
 namespace System.Net.Http
 {
     public class HttpMethod : IEquatable<HttpMethod>
     {
-        private string _method;
-
-        private static readonly HttpMethod s_getMethod = new HttpMethod("GET");
-        private static readonly HttpMethod s_putMethod = new HttpMethod("PUT");
-        private static readonly HttpMethod s_postMethod = new HttpMethod("POST");
-        private static readonly HttpMethod s_deleteMethod = new HttpMethod("DELETE");
-        private static readonly HttpMethod s_headMethod = new HttpMethod("HEAD");
-        private static readonly HttpMethod s_optionsMethod = new HttpMethod("OPTIONS");
-        private static readonly HttpMethod s_traceMethod = new HttpMethod("TRACE");
-
-        // Don't expose CONNECT as static property, since it's used by the transport to connect to a proxy. 
+        private readonly string _method;
+        private int _hashcode;
```

Comment:
This optimization assumes that saving the string allocation in GetHashCode is worth adding an extra field in HttpMethod and thus increasing the size of the HttpMethod object.  I don't know enough about the use cases for HttpMethod to know whether that's a net win... I assume yes, but just verifying...?